ckanext/gdi_userportal/logic/action/get.py

```diff
@@ -47,41 +72,88 @@ def scheming_package_show(context, data_dict):
 
-        if ('display_snippet' in field and field['display_snippet'] in [None, 'hidden.html']) \
-                or field_name in fields_in_schema_that_should_not_be_visible:
+        if (
+            "display_snippet" in field
+            and field["display_snippet"] in [None, "hidden.html"]
+        ) or field_name in fields_in_schema_that_should_not_be_visible:
             visible = False
         else:
             visible = True
 
-        field_label = field.get('label', None)
+        field_label = field.get("label", None)
         choices = toolkit.h.scheming_field_choices(field)
 
         if choices is None:
-            new_pkg_dict[field_name] = {"value": value, "value_label": None,
-                                        "field_label": field_label, "visible": visible}
+            new_pkg_dict[field_name] = {
+                "value": value,
+                "value_label": None,
+                "field_label": field_label,
+                "visible": visible,
+            }
             del pkg_dict[field_name]
             continue
 
         for choice in choices:
-            if choice['value'] == value:
-                new_pkg_dict[field_name] = {"value": value, "value_label": choice['label'],
-                                            "field_label": field_label, "visible": visible}
+            if choice["value"] == value:
+                new_pkg_dict[field_name] = {
+                    "value": value,
+                    "value_label": choice["label"],
+                    "field_label": field_label,
+                    "visible": visible,
+                }
                 del pkg_dict[field_name]
                 break
 
     for fieldname in pkg_dict:
         field_label = None
-        visible = any(fieldname in field_dict['name'] for field_dict in fields_not_in_schema_that_should_be_visible)
+        visible = any(
+            fieldname in field_dict["name"]
+            for field_dict in fields_not_in_schema_that_should_be_visible
+        )
         if visible:
             for field_dict in fields_not_in_schema_that_should_be_visible:
-                if field_dict['name'] == fieldname:
-                    field_label = field_dict['label']
-
-        default_field_dict = {"value": pkg_dict[fieldname],
-                              "value_label": None,
-                              "field_label": field_label,
-                              'visible': visible
-                              }
+                if field_dict["name"] == fieldname:
+                    field_label = field_dict["label"]
+
+        default_field_dict = {
+            "value": pkg_dict[fieldname],
+            "value_label": None,
+            "field_label": field_label,
+            "visible": visible,
+        }
         new_pkg_dict[fieldname] = default_field_dict
 
     return new_pkg_dict
+
+
+@toolkit.side_effect_free
+def get_keyword_list(context, data_dict) -> dict[str, int | list[str]]:
```

Comment:
Is it possible to combine the several endpoints to reduce network traffic?

Reply:
I am not sure merging the endpoints to a single one will help. 

The different queries are fired up simultaneously and run in parallel. With a single endpoint, we would loose that while keeping the data size to be transferred. What do you think?

Reply:
I have not lot experience with solving this in python, but in .NET I was used to solve those tasks asynchronous